app/Observers/UserObserver.php

```diff
@@ -0,0 +1,21 @@
+<?php
+namespace App\Observers;
+
+use App\User;
+use Illuminate\Support\Facades\Hash;
+
+class UserObserver {
+
+  public function created(User $user) {
```

Comment:
Both on a new user being created and a user being updated will a "saving" event fire, why not have just one function registered for saving that does this instead of two?

https://laravel.com/docs/5.4/eloquent#events

Whenever a new model is saved for the first time, the creating and created events will fire. If a model already existed in the database and the save method is called, the updating / updated events will fire. However, in both cases, the saving / saved events will fire.